janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -100,9 +100,11 @@ public String toString() {
 
     @Override
     public JointIndexQuery updateLimit(int newLimit) {
-        JointIndexQuery ji = new JointIndexQuery(Lists.newArrayList(queries));
-        ji.setLimit(newLimit);
-        return ji;
+        List<Subquery> subqueries = (queries.size() == 1) ? Lists.newArrayList(queries.get(0).updateLimit(newLimit)) :
+            Lists.newArrayList(queries);
+        JointIndexQuery jointIndexQuery = new JointIndexQuery(subqueries);
```

Comment:
Hi @mad!  could you explain why we need:
1. Copy `queries` into a new ArrayList,
and
2. Create a new JointIndexQuery object?

Does an in-place update not work? Thanks!

Reply:
Old implementation does same things but without update single query limit

And doc for `BackendQuery.updateLimit` sad
```
Creates a new query identical to the current one but with the specified limit.
```

So this method create new one query and update limit. Some `BaseQuery` implementation does not support limit update

Reply:
Thanks @mad ! I didn't notice the contract written in `BackendQuery.updateLimit`. I took a quick look and it seems all current implementations support limit update. IMO we could change the contract and always use an in-place update, which could save a little bit of object copy time. I would be happy to contribute a patch if applicable.

Reply:
I took a closer look, and you're right, there are some implementations that don't support in-place update. And there are some use cases where a new query is preferable. So I guess we shall leave them as they are.